server/backends/invocationdb/invocationdb.go

```diff
@@ -88,9 +90,18 @@ func (d *InvocationDB) registerInvocationAttempt(ctx context.Context, ti *tables
 
 func (d *InvocationDB) CreateInvocation(ctx context.Context, ti *tables.Invocation) (bool, error) {
 	var permissions *perms.UserGroupPerm
+	var caps []akpb.ApiKey_Capability
 	if auth := d.env.GetAuthenticator(); auth != nil {
-		if u, err := auth.AuthenticatedUser(ctx); err == nil && u.GetGroupID() != "" {
-			permissions = perms.GroupAuthPermissions(u.GetGroupID())
+		u, err := auth.AuthenticatedUser(ctx)
+		if err == nil {
+			if u.GetGroupID() != "" {
+				permissions = perms.GroupAuthPermissions(u.GetGroupID())
+			}
+			caps = u.GetCapabilities()
+		} else {
+			if perms.IsAnonymousUserError(err) && auth.AnonymousUsageEnabled() {
```

Comment:
@bduffany Realized I didn't handle anonymous users. Can you verify that it's the intended behavior here to use the default capabilities (includes write) for anonymous users?

Reply:
Nice catch -- yep, this looks right.